code/modules/recycling/conveyor2.dm

```diff
@@ -125,43 +124,39 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		return FALSE
 	return TRUE
 
-// returns true if there is at least 1 item on the belt that can be moved
-/obj/machinery/conveyor/proc/check_belt_contents()
-	for(var/atom/movable/AM in loc)
-		if(!AM.anchored)
-			return TRUE
-	return FALSE
+// machine process
+// move items to the target location
+/obj/machinery/conveyor/process()
+	if(machine_stat & (BROKEN | NOPOWER))
+		return
+
+	//If the conveyor is broken or already moving items
+	if(!operating || conveying)
+		return
 
-// populates `affecting` with the first 30 items in the belt's contents
-/obj/machinery/conveyor/proc/get_belt_contents()
+	use_power(6)
+
+	//get the first 30 items in contents
 	affecting = list()
 	var/i = 0
-	for(var/atom/movable/AM in loc)
-		if(AM.anchored) // If it can't be moved, ignore it
+	for(var/item in loc.contents)
+		if(item == src)
 			continue
-		if(++i > MAX_CONVEYOR_ITEMS_MOVE)
+		i++ // we're sure it's a real target to move at this point
+		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
 			break
-		affecting.Add(AM)
-
-// when some item or mob moves over the conveyor belt
-/obj/machinery/conveyor/proc/conveyorCrossed(datum/source, atom/movable/AM, atom/oldLoc)
-	// if the conveyor isn't switched on or is broken / has no power
-	if(!operating || machine_stat & (BROKEN | NOPOWER))
-		return
+		affecting.Add(item)
 
-	get_belt_contents() // populate affecting with the movable items on the belt
-	addtimer(CALLBACK(src, .proc/convey, affecting), pick(1,2)) // pick(1,2) so players have a bit of trouble walking over belts moving in the opposite direction they are
+	conveying = TRUE
+	addtimer(CALLBACK(src, .proc/convey, affecting), 1)
 
-// pushes the items sitting on the belt in the direction the belt is operating in
 /obj/machinery/conveyor/proc/convey(list/affecting)
 	for(var/atom/movable/A in affecting)
 		if(!QDELETED(A) && (A.loc == loc))
 			A.ConveyorMove(movedir)
 			//Give this a chance to yield if the server is busy
 			stoplag()
```

Comment:
I think `if(!QDELETED(A) && (A.loc == loc))` can be removed if use `for(var/atom/movable/A in affecting&loc.contents)`
QDELETED items must be in nullspace so it no in contents of our turf.

```suggestion
	for(var/atom/movable/A in affecting&loc.contents)
		A.ConveyorMove(movedir)
		//Give this a chance to yield if the server is busy
		stoplag()
```


As i think about this. 
It need check like "its posible to move something?" before we try to move all things, 
or we create load generator,
 by place full conveyor directed to wall.
But while we has limit to 30 things its must be small load.